build_tools/azure/get_commit_message.py

```diff
@@ -0,0 +1,27 @@
+import os
+import subprocess
+
+
+def get_commit_message():
+    """Retrieve the commit message."""
+    build_source_version_message = os.environ["BUILD_SOURCEVERSIONMESSAGE"]
+
+    if os.environ["BUILD_REASON"] == "PullRequest":
+        # By default pull requests use refs/pull/PULL_ID/merge as the source branch
+        # which has a "Merge ID into ID" as a commit message. The latest commit
+        # message is the second to last commit
+        commit_id = build_source_version_message.split()[1]
+        git_cmd = f"git log {commit_id} -1 --pretty=%B"
+        commit_message = subprocess.run(
+            git_cmd, shell=True, capture_output=True, text=True
+        ).stdout.strip()
```

Comment:
I think we can write this the following to avoid `shell=True`:

```python
commit_message = subprocess.run(
    ["git", "log", commit_id, "-1", "--pretty=%B"],
    text=True, capture_output=True).stdout.strip()
```

This also avoids the security issue with `shell=True`: https://docs.python.org/3/library/subprocess.html#security-considerations